std/math.d

```diff
@@ -6626,3 +6627,250 @@ real yl2xp1(real x, real y) @nogc @safe pure nothrow;       // y * log2(x + 1)
     auto x = floor(1.2);
     auto y = ceil(1.2);
 }
+
+/***********************************
+ * Defines a total order on all floating-point numbers.
+ *
+ * The order is defined as follows:
+ * $(UL
+ *      $(LI All numbers in [-$(INFIN), +$(INFIN)] are ordered
+ *          the same way as by built-in comparison, with the exception of
+ *          -0.0, which is less than +0.0;)
+ *      $(LI If the sign bit is set (that is, it's 'negative'), $(NAN) is less
+ *          than any number; if the sign bit is not set (it is 'positive'),
+ *          $(NAN) is greater than any number;)
+ *      $(LI $(NAN)s of the same sign are ordered by the payload ('negative'
+ *          ones - in reverse order).)
+ * )
+ *
+ * Returns:
+ *      -1 if $(D x) precedes $(D y) in the order specified above;
+ *      0 if $(D x) and $(D y) are identical, and 1 otherwise.
+ *
+ * See_Also:
+ *      $(MYREF isIdentical)
+ * Standards: Conforms to IEEE 754-2008
+ */
+int cmp(T)(T x, T y) @nogc @trusted pure nothrow
+    if (isFloatingPoint!T)
+{
+    alias F = floatTraits!T;
+
+    static if (F.realFormat == RealFormat.ieeeSingle
+               || F.realFormat == RealFormat.ieeeDouble)
+    {
+        static if (T.sizeof == 4)
+        {
+            alias UInt = uint;
+        }
+        else
+        {
+            alias UInt = ulong;
+        }
+
+        enum msb = ~(UInt.max >>> 1);
+
+        UInt*[2] vars = [cast(UInt*)&x, cast(UInt*)&y];
+
+        foreach (var; vars)
+        {
+            if (*var & msb)
+            {
+                *var = ~*var;
+            }
+            else
+            {
+                *var |= msb;
+            }
+        }
+
+        return *vars[0] < *vars[1] ? -1 : *vars[0] == *vars[1] ? 0 : 1;
+    }
+    else static if (F.realFormat == RealFormat.ieeeExtended53
+                    || F.realFormat == RealFormat.ieeeExtended
+                    || F.realFormat == RealFormat.ieeeQuadruple)
+    {
+        static if (F.realFormat == RealFormat.ieeeQuadruple)
+        {
+            alias RemT = ulong;
+        }
+        else
+        {
+            alias RemT = ushort;
+        }
+
+        enum shift = ulong.sizeof / RemT.sizeof;
+
+        ubyte*[2] bytes = [cast(ubyte*)&x, cast(ubyte*)&y];
+
+        ulong*[2] bulk = [cast(ulong*)&x, cast(ulong*)&y];
+        RemT*[2] rem = [cast(ushort*)&x + shift, cast(ushort*)&y + shift];
+
+        foreach (i; 0 .. 2)
+        {
+            if (bytes[i][F.SIGNPOS_BYTE] & 0x80)
+            {
+                *bulk[i] = ~*bulk[i];
+                *rem[i] = ~*rem[i];
+            }
+            else
+            {
+                bytes[i][F.SIGNPOS_BYTE] |= 0x80;
+            }
+        }
+
+        version(LittleEndian)
+        {
+            alias major = rem;
+            alias minor = bulk;
+        }
+        else
+        {
+            alias major = bulk;
+            alias minor = rem;
+        }
+
+        return *major[0] < *major[1]
```

Comment:
So, looking back at this, I have no idea how I failed to notice this. I would suggest an if-else if chain rather than this, as it is much clearer with no difference in performance.